superset/db_engine_specs/snowflake.py

```diff
@@ -128,3 +129,34 @@ def mutate_db_for_connection_test(database: "Database") -> None:
         engine_params["connect_args"] = connect_args
         extra["engine_params"] = engine_params
         database.extra = json.dumps(extra)
+
+    @classmethod
+    def get_cancel_query_id(cls, cursor: Any, query: Query) -> Optional[str]:
+        """
+        Get Snowflake session ID that will be used to cancel all other running
+        queries in the same session.
+
+        :param cursor: Cursor instance in which the query will be executed
+        :param query: Query instance
+        :return: Snowflake Session ID
+        """
+        cursor.execute("SELECT CURRENT_SESSION()")
```

Comment:
Does this work? The reason I ask is per the documentation, [CURRENT_SESSION](https://docs.snowflake.com/en/sql-reference/functions/current_session.html)

> Returns a unique system identifier for the Snowflake session corresponding to the present connection

yet the connection associated with said cursor is not the same connection which instantiated the query (in all likelihood for async queries that connection resides on a Celery host). Thus I would suspect there are actually no queries running within the current session.

Reply:
The query that you're referring to is running when the query initiated and not when the stop button pressed.
It's using the same cursor that is created to run the query in the `execute_sql_statements` [here](https://github.com/koszti/superset/blob/517c7928dfee2412b96836957a2efd2b5953c1e5/superset/sql_lab.py#L452). The returned session id then saved into `query` table in the superset backend database.

When the stop button pressed, we get the saved session id from the `query` table and kill it and kill the query by another SQL. In this way queries can be killed at a later from every celery worker.

It's tested on Postgres, MySQL and Snowflake engines with sync and async query mode.

Reply:
Apologies. I misread the code logic.